### DISCUSSION TITLE
Using Ubuntu to cross-compile for Windows

### DISCUSSION THREAD

**Discussion by js333031** [microsoft/vcpkg#42900] [General]
Hello, is my understanding correct that Linux host can be used to compile Windows applications? I would like to compile digikam but when I run 01-build-vcpkg.sh (located here) on Ubuntu 24.04.1 LTS, I get following error:
`error: in triplet x64-windows: Use of Visual Studio's Developer Prompt is unsupported on non-Windows hosts.`

I will also post this question on digikam's repo.

**Comment by js333031** [comment]
If there is a different triplet for cross-compilation of Windows apps on Linux hosts, please let me know. I've also tried x86-mingw-dynamic

**Reply by msclock** [reply]
I used to manage to cross-compile with x64-mingw-dynamic for windows on ubuntu on the [old hash of ss-cpp](https://github.com/serious-scaffold/ss-cpp/tree/0ecd63eb80067e5ab2055479af5b9c5e724893ef). 


There are bunch of presets and cmake tricks and believe me it's not a confortable experience.

**Reply by dg0yt** [reply]
> There are bunch of presets and cmake tricks and believe me it's not a confortable experience.

This is useless information. It is unclear what you tried to build, and what kind of problems you encountered.

**Comment by dg0yt** [comment]
> x64-windows

The "windows" triplets are for MSVC. When you want to cross compile for Windows on Ubuntu, you probaly want to use "mingw", e.g. x64-mingw-static or x64-mingw-dynamic.

**Reply by js333031** [reply]
Thank you for suggesting x64-mingw-dynamic. I tried it and ran into an error for which I submitted following issue: https://github.com/microsoft/vcpkg/issues/42993
